src/wallet/wallet.cpp

```diff
@@ -4257,6 +4257,9 @@ util::Result<MigrationResult> MigrateLegacyToDescriptor(const std::string& walle
         success = local_wallet->IsWalletFlagSet(WALLET_FLAG_BLANK_WALLET);
         if (!success) {
             success = DoMigration(*local_wallet, context, error, res);
+        } else {
+            // Make sure that descriptors flag is actually set
+            local_wallet->SetWalletFlag(WALLET_FLAG_DESCRIPTORS);
```

Comment:
Given that the PR describes the setting of the flag as a special case for blank wallets, I’d be more comfortable if the setting of the flag were conditional on a wallet being blank instead of success being true here.

```diff
-        success = local_wallet->IsWalletFlagSet(WALLET_FLAG_BLANK_WALLET);
+        if(local_wallet->IsWalletFlagSet(WALLET_FLAG_BLANK_WALLET)) {
+            local_wallet->SetWalletFlag(WALLET_FLAG_DESCRIPTORS);
+            success = true;
+        }
+
        if (!success) {
            success = DoMigration(*local_wallet, context, error, res);
-        } else {
-            // Make sure that descriptors flag is actually set
-            local_wallet->SetWalletFlag(WALLET_FLAG_DESCRIPTORS);
```

Reply:
> I’d be more comfortable if the setting of the flag were conditional on a wallet being blank instead of success being true

Just to be clear, these two conditions should be exactly equivalent due to the `success = local_wallet->IsWalletFlagSet(WALLET_FLAG_BLANK_WALLET);` line so should be a style change, not a change in behavior. I agree the suggestion looks a little better, but the style of dealing with the `success` variable seems here to follow a particular pattern, so would leave it up to achow to decide which way fits better.

Reply:
I think it should actually unconditionally set the flag, but I haven't fully thought through if that is safe yet. Intuitively, it should be safe since a failed migration would result in the wallet being deleted anyways, but I haven't considered all of the possibilities yet.

Reply:
@ryanofsky Yeah, I was not proposing a change in behavior. It just seems cleaner to tie the intended behavior directly to the relevant wallet property than to a generic boolean like `success` that is instantiated half a code block away and much more likely to change its meaning in the course of the project.

@achow101: I would have expected that `DoMigration` sets `WALLET_FLAG_DESCRIPTOR`, but it doesn’t seem to be the case. If you wanted to set it generally, perhaps you could just set it after the `if (!success) {…}` block here?

Reply:
`DoMigration` does do that, that's why this is only a problem for blank wallets.

As currently implemented, this only sets it for blank wallets, but I do also think the correct behavior is to set it for all wallets, but as I said earlier, I have not analyzed that yet.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/29367#discussion_r1476516869

> `DoMigration` does do that, that's why this is only a problem for blank wallets.

Seems like in a followup it might be cleaner to do this inside DoMigration, so the flag is set in one place. In the meantime it could help to change the "Make sure that descriptors flag is actually set" comment to "Make sure that descriptors flag is set if DoMigration is not called (otherwise DoMigration will set it)"